src/main/drivers/pwm_output.c

```diff
@@ -359,6 +369,22 @@ void pwmCompleteMotorUpdate(void)
 
 #ifdef USE_DSHOT
     if (isMotorProtocolDshot()) {
+
+        //Check if there are commands to send
+        if(!circularBufferIsEmpty(&commandsCircularBuffer) &&
+            micros() - lastCommandSent > DSHOT_COMMAND_INTERVAL_US){
+
+            dshotCommands_e cmd;
+            circularBufferPopHead(&commandsCircularBuffer,(uint8_t*) &cmd);
+
+            for (uint8_t i = 0; i < getMotorCount(); i++) {
+                pwmRequestMotorTelemetry(i);
```

Comment:
No need to use abstractions - we know what data values we have to update so we can update `motors[index].value` and `motors[index].requestTelemetry` directly.